tests/test_other.py

```diff
@@ -10630,3 +10630,15 @@ def test_no_deprecated(self):
   def test_bad_export_name(self):
     err = self.expect_fail([EMCC, '-sEXPORT_NAME=foo bar', test_file('hello_world.c')])
     self.assertContained('error: EXPORT_NAME is not a valid JS identifier: `foo bar`', err)
+
+  def test_standard_library_mapping(self):
+    # Test the `-l` flags on the command line get mapped the correct libraries variant
+    self.run_process([EMBUILDER, 'build', 'libc-mt', 'libcompiler_rt-mt', 'libdlmalloc-mt'])
```

Comment:
Why is the embuilder line needed? (auto library building should happen on line 10639?)

lgtm without line 10636

Reply:
Automatic building currently only works for libraries that are added automatically in `system_libs.py:calculate()`.

Libraries specified on the command line are not currently auto-built.  This is just how its always been as far as I cant tell.  We could change that but its what this change is about.

Also, `-nostdlib` essentially bypasses `system_libs.py:calculate()` and doesn't do any auto-adding or auto-building of libraries.

This PR is an incremental improvement.  We could integrating `-l` flags with the auto-building system but that would be bigger/separate change.